tests/addons/test_addon.py

```diff
@@ -97,6 +98,27 @@ def test_additional_catalogue_src_name_validation():
         addon._validate_additional_catalogue_srcs(metadata)
 
 
+def test_secret_names_validation():
+    addon = Addon(addon_with_secrets_path(), "stage")
+    metadata = addon.metadata
+    duplicate = metadata["secrets"][0]
+    metadata["secrets"].append(duplicate)
+    with pytest.raises(AddonLoadError):
+        addon._validate_secret_names(metadata)
+
+
+def test_pullSecretName_validation():
+    addon = Addon(addon_with_secrets_path(), "stage")
+    metadata = addon.metadata
+    pullSecretName = metadata["pullSecretName"]
+    secrets = metadata["secrets"]
+    metadata["secrets"] = list(
+        filter(lambda secret: secret["name"] != pullSecretName, secrets)
```

Comment:
```suggestion
    addon.metadata["pullSecretName"] = "foo1234"
```
Instead of modifying `secrets`, let's just modify pullSecretsName in such a way that it doesn't belong to `secrets`. Cleaner and faster :)